docs(content-releases): add frontend docs (#19391)
strapi/strapi: docs/docs/docs/01-core/content-releases/02-frontend/02-details-view.mdx

---
title: Release details view
tags:
  - content-releases
  - tech design
---

### Overview:

> The details-view offers a comprehensive presentation of all entries linked to the release. These entries are meticulously organized into tables based on various options. Each entry's status is clearly indicated, providing the ability to edit or remove entries from the release.

Reviewer: ```suggestion
The details-view offers a comprehensive presentation of all entries linked to the release. The entries are grouped into sections of tables displaying useful information about each entry such as its locale, the action to take (publish/unpublish), the status of the entry (published, unpublished, or validation errors).
```

> On this page, users with the appropriate permissions can perform key actions such as publishing the release, editing its details, or deleting it.

Reviewer: ```suggestion
On this page, users with the appropriate permissions can do the following:
```

> #### Grouping entries:

Reviewer: ```suggestion
#### Group entries:
```

Entries associated with a release can be systematically organized in tables using various grouping options:

> - <strong>by Content-Types</strong>: Organize entries by collections.

Reviewer: I wouldn't say by collections since single types are not a collection

- <strong>by Locales</strong>: Group entries based on locale.
- <strong>by Actions</strong>: Categorize entries by action to perform, making it easier to discern
  between entries to publish and entries to unpublish.

#### Edit a release:

> Users have the flexibility to change the name of a release, by simply clicking on the "More" button, the Edit button is revealed (enabled if the user role has `plugin::content-releases.update` permission).

Reviewer: I don't think we should call it a more button since that is not how it is represented by text in the UI or by accessible name.

Reviewer: Maybe something along the lines of:
```suggestion
If the user has the `plugin::content-releases.update` permissions, a "three dot" button will appear in the header giving access to a menu with an "Edit" option.  
```

The Edit button opens a modal, enabling users to modify the release name. To save the changes, the new name must be unique, non-empty, and distinctly modified.

#### Delete a release:

Deleting a release is a straightforward process for users. By clicking on the "More" button, the "Delete" button is exposed (enabled if the user role has `plugin::content-releases.delete` permission).

> Upon activation, the "Delete" button triggers a confirmation modal, prompting the user to confirm the deletion.

Reviewer: Same as above and I think the action is very similar to the edit one so maybe the description here could be as similar.

> #### Managing Entry Status in a Release:

Reviewer: ```suggestion
#### View Entry Status
```

> Each entry within a release is assigned a status, indicating its readiness for actions such as Publish/Unpublish or any validation errors present. In cases where errors are detected, users can navigate to the "Three dots" menu and select the "Edit entry" button to access the CM edit view.

Reviewer: ```suggestion
Each entry within a release is assigned a status, indicating its readiness for actions such as Publish/Unpublish or any validation errors present. If the entry has a validation errors. A user can use the entry's "Three dots" menu to access a "Edit entry" link to the CM edit view.
```

> Within the CM edit view, users can resolve any identified errors. Upon completion, clicking the "Refresh" button updates the status of every individual entry, reflecting the changes made.

Reviewer: ```suggestion
Within the CM edit view, users can resolve any identified errors. Upon completion, clicking the "Refresh" button on the release details page updates the status of each entry to reflect any changes made.
```
